src/fides/api/models/property.py

```diff
@@ -60,21 +74,82 @@ def create(
         check_name: bool = True,
     ) -> Property:
         experiences = data.pop("experiences", [])
+        paths = data.pop("paths", None) or []
+
+        matching_paths = PropertyPath.filter(
+            db=db, conditions=(PropertyPath.path.in_(paths))
+        ).all()
+        if matching_paths:
+            raise KeyOrNameAlreadyExists(
+                f'The path(s) \'{", ".join([matching_path.path for matching_path in matching_paths])}\' are already associated with another property.'
+            )
+
```

Comment:
very nice! as a nit, i might move this into a helper function to share between create and update in case the logic ever needs to be adjusted, but i'll let you decide whether or not to add that last bit of polish :)